modules/channel-twilio/src/backend/index.ts

```diff
@@ -0,0 +1,78 @@
+import * as sdk from 'botpress/sdk'
+import { Config } from 'src/config'
+
+import { removeMiddleware, setupMiddleware, TwilioClient } from './client'
+import { Clients } from './typings'
+
+let router: sdk.http.RouterExtension
+const route = '/webhook'
+const clients: Clients = {}
+
+const onServerStarted = async (bp: typeof sdk) => {
+  await setupMiddleware(bp, clients)
+}
+
+const onServerReady = async (bp: typeof sdk) => {
+  router = bp.http.createRouterForBot('channel-twilio', {
+    checkAuthentication: false
+  })
+
+  router.post(route, async (req, res) => {
+    const { botId } = req.params
+    const client = clients[botId]
+
+    if (client) {
+      if (client.auth(req)) {
```

Comment:
Can you implement this as a middleware instead? It would be cleanlier

Reply:
I'm not sure if that's what you meant but I moved the logic in a separate function outside of `index.ts`.

Reply:
Check this file for an implementation idea: 
https://github.com/botpress/botpress/blob/master/modules/code-editor/src/backend/utils_router.ts#L57